components/sidebar/sidebar.jsx

```diff
@@ -199,7 +199,7 @@ export default class Sidebar extends React.PureComponent {
             this.channelIdIsDisplayedForProps(prevProps.orderedChannelIds, this.props.currentChannel.id)
         ) {
             this.closedDirectChannel = true;
-            browserHistory.push(`/${this.props.currentTeam.name}/channels/${Constants.DEFAULT_CHANNEL}`);
```

Comment:
My concenr here is: this code is designed to be called when a direct channel autocloses, looks like somehow we are reusing this side-effect to do something, or we are calling it accidentally when you leave the team.

If we change this, maybe when a channel "autocloses" it will redirect you to your previous team/previous channel.

Reply:
Is that what the code is designed for? The comment seemed to state it will do that for any active channel that no longer exists, with autoclosed DMs just being one example. I admit I don't like performing any type of redirection in a `componentDidUpdate` function but I was worried about changing too much here and breaking cases I haven't thought of

Reply:
To be clear, this redirect isn't required when leaving a team. There's another redirect that handles that but this side effect one was causing the problem.

Reply:
Yes, I understand that, I'm actually working with the redirections to default channel, and is code that is very difficult to predict the proper behavior.

Anyway, I think you are right, anything that generates a "dissapear" of the channel can trigger this, for example, a delete channel event from other user. But still, is the same problem, you will be redirected to defaultTeam+defaultChannel, not to the defaultChannel in the current team. I think probably. My fear is to remove one bug and add another.

Reply:
It shouldn't redirect you to a different team then the one you're on. If you look here you'll see that we set the "previous" team ID to your current team ID https://github.com/mattermost/mattermost-webapp/blob/master/components/needs_team/needs_team.jsx#L182

Reply:
@jwilander 
https://github.com/mattermost/mattermost-webapp/blob/master/actions/websocket_actions.jsx#L709-L711 seems to be doing the redirect when a channel is left so do we need to do this? 
I think doing this on action level and removing these LOC here can help.

I am 0/5 on having this change here but if we don't need them because of the above those lines we can remove them

Reply:
As I said above, I know this isn't needed for leaving a team or channel. This is needed by the specific case the code comment mentions for auto-closed DMs but handling that case with a `browserHistory.push` like we were was causing a problem when leaving teams. So we can't just remove this line or that case will break. I'm open to suggestions on how to refactor to handle that case better, as I really don't like that we're redirecting in a componentDidUpdate function, but I can't think of a good fix that doesn't require a ton of mana and I'm pretty confident that my proposed fix works

@sudheerDev @jespino

Reply:
I'm referring to this comment https://github.com/mattermost/mattermost-webapp/blob/311bcba24cdcb00354e46d8e5640d632a7533603/components/sidebar/sidebar.jsx#L195

Reply:
Aah got it make sense now. I am good with the change now. I tested it and works as described.
FYI if we run into other issues i think we should remove this and rather depend on

<img width="1130" alt="Screenshot 2019-03-26 at 10 38 32 PM" src="https://user-images.githubusercontent.com/4973621/55019150-5a8be480-501a-11e9-80c5-647ba3ee03dc.png">

socket event for redirection